viam-cartographer/src/carto_facade/carto_facade.cc

```diff
@@ -264,8 +266,13 @@ void CartoFacade::IOInit() {
     }
     // Setup file system for saving internal state
     setup_filesystem(config.data_dir, path_to_internal_state);
-    slam_mode =
-        determine_slam_mode(path_to_internal_state, config.map_rate_sec);
+    if (config.cloud_story_enabled == true) {
```

Comment:
The new flag that was added in this PR is used to determine the slam mode instead of `map_rate_sec` so that in the next PR we can stop using `map_rate_sec` in cloud story
```
if (!enable_mapping) {
                // This log line is needed by rdk integration tests.
                LOG(INFO) << "Running in localization only mode";
                return viam::carto_facade::SlamMode::LOCALIZING;
            }

```

Reply:
should we just have one function and pass in the feature flag? That way we don't need the internal state loop in two spots, and I think would make removing the feature flag simpler down the line

Reply:
I think this makes both clean-up and understanding where the cloud story changes need to go a bit cleaner. This way during cleanup we can delete the entire `determine_slam_mode` function and associated tests & as we make the additional changes for determining slam mode im cloud story it is very obvious what function needs to be changed and less likely to break old cartographer. What do other people think?

Reply:
I agree with kim.